[jit] remove remnants of properties hack

suo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #31020 [jit] avoid doing quadratic work in concrete type inference
* #31019 [jit] Simplify recursive script compilation flow.
* **#31018 [jit] remove remnants of properties hack**
* #31017 [jit] remove unnecessary arg from create_script_module

Properties are now disallowed so this hack is no longer necessary

Differential Revision: [D18904112](https://our.internmc.facebook.com/intern/diff/D18904112)

suo committed:
[jit] remove remnants of properties hack

Properties are now disallowed so this hack is no longer necessary

kostmo commented:
## CircleCI build failures summary
As of commit 0ec3023:
* **1/1** failures introduced in this PR



## Detailed failure analysis (WIP)

Here are the [reasons each build failed](https://dr.pytorch.org/commit-details.html?sha1=0ec30231faf246d27ad54e902993efb64eec393e):

### 1 failure recognized by patterns:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3873186) pytorch-linux-xenial-py3-clang5-android-ndk-r19c-gradle-build (1/1)
**Step:** pytorch android gradle build ([details](https://dr.pytorch.org/build-details.html?build_id=71131808))
```
Error response from daemon: manifest for 308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/pytorch-linux-xenial-py3-clang5-android-ndk-r19c:405-0ec30231faf246d27ad54e902993efb64eec393e-android-arm-v7a not found
```


---
This comment was automatically generated by [Dr. CI](https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment).
Follow [this link to opt-out](https://dr.pytorch.org/admin/comments-opt-out.html) of these comments for your Pull Requests.


Please report bugs/suggestions on the [GitHub issue tracker](https://github.com/kostmo/circleci-failure-tracker/issues).

suo committed:
Update on "[jit] remove remnants of properties hack"


Properties are now disallowed so this hack is no longer necessary

Differential Revision: [D18904112](https://our.internmc.facebook.com/intern/diff/D18904112)
